olp-cpp-sdk-core/include/olp/core/porting/deprecated.h

```diff
@@ -0,0 +1,29 @@
+/*
+ * Copyright (C) 2019 HERE Europe B.V.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ * License-Filename: LICENSE
+ */
+
+#pragma once
+
+#if defined(__GNUC__) || defined(__clang__)
+#define DEPRECATED __attribute__((deprecated))
```

Comment:
```suggestion
#define OLP_SDK_DEPRECATED(text) __attribute__((deprecated))
```
Also add text (we dont need to use it) but then the users see a text:

```cpp
OLP_SDK_DEPRECATED("Will be removed by 11.2019, please use GetData(request, callback) instead")
ErrorCode GetData(callback, request);
```